trustfall_core/src/graphql_query/error.rs

```diff
@@ -88,6 +88,13 @@ pub enum ParseError {
     #[error("Edge {1} specifies a duplicated parameter {0}")]
     DuplicatedEdgeParameter(String, String, Pos),
 
+    #[error(
+        "Found GraphQL query variable definition(s). \
+    These are not necessary since variables are defined implicitly, \
+    and must be removed."
+    )]
+    VariableDefinitionInRootQuery(Pos),
```

Comment:
I'm confused by the "root query" bit, since there's only one query — we error if the user tries to send more than one query in one document.

I'm curious to hear more about what you had in mind with the name. Also happy to do a quick rename as well:
```suggestion
    VariableDefinitionInQuery(Pos),
```

Reply:
Ah you're correct - I was sort of following the naming scheme for other variables within the area of code coverage haha. I'll rename.